components/wmtk_components/marching/internal/Marching.hpp

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include <wmtk/Scheduler.hpp>
+#include <wmtk/TriMesh.hpp>
+
+namespace wmtk::components::internal {
+
+class Marching
+{
+    TriMesh& m_mesh;
+    bool m_lock_boundary;
+
+    MeshAttributeHandle<double> m_position_handle; // record position
+    MeshAttributeHandle<long> m_vertex_tag; // record vertex tag value
+    MeshAttributeHandle<long> m_edge_tag; // record edge vertex tag value
+    long m_input_tag_value; // the value used to those simplicity you want to seperate
+    long m_embedding_tag_value; // the scalffold simplicities' tag value
+    long m_split_tag_value; // when you split a simplicity, you will set m_split_tag_value to the
```

Comment:
This component should create an isosurface. Therefore, all that is required is the isovalue.

In your case for example, you might have an embedding tag of 2 and an input tag of 0. Then you can set your isovalue to 1. Every edge whose vertices have one tag above 1 and one tag below 1 are then split. The new vertex gets then the tag 1.